metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -302,3 +385,185 @@ func getHostSystem(ctx context.Context, c *vim25.Client, ref types.ManagedObject
 	}
 	return &hs, nil
 }
+
+func splitIntoChunks(slice []types.ManagedObjectReference, n int) [][]types.ManagedObjectReference {
+	var chunks [][]types.ManagedObjectReference
+
+	for i := 0; i < len(slice); i += n {
+		end := i + n
+
+		// Check if the end index is beyond slice bounds
+		if end > len(slice) {
+			end = len(slice)
+		}
+
+		chunks = append(chunks, slice[i:end])
+	}
+
+	return chunks
+}
+
+func getVmFilter(pools []types.ManagedObjectReference) property.Filter {
+	return property.Filter{"resourcePool": pools}
+}
+
+func (m *MetricSet) getVirtualMachineList(ctx context.Context, c *vim25.Client, mgr *view.Manager, includedPools []string) ([]mo.VirtualMachine, error) {
+
+	v, err := mgr.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"VirtualMachine"}, true)
+	if err != nil {
+		return nil, fmt.Errorf("error in CreateContainerView: %w", err)
+	}
+
+	defer func() {
+		if err := v.Destroy(ctx); err != nil {
+			m.Logger().Debug(fmt.Errorf("error trying to destroy view from vshphere: %w", err))
```

Comment:
no need to wrap the error here also using just Debugf here:
```suggestion
			m.Logger().Debugf("error trying to destroy view from vshphere: %s", err)
```

Reply:
fixed.